### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by ludziej**
# Description
Itegration tests that run toy model on CPU, within unit tests

# Checklist
Answer each question with "yes", and provide an explanation if the answer isn't what is expected.

1. Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - Answer: y
2. Are all variables set with logical defaults that are backward compatible?
   - Answer: y
3. Can you provide a link to a Neptune experiment verifying the functionality of this PR?
   - Answer: y
4. Have you successfully executed the linter and tests?
   - Answer: y
5. Are all functions concise and don't require splitting?
   - Answer: y
6. Is there documentation for arguments and complex logic?
   - Answer: y
7. Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - Answer: y
8. Are all functions placed in appropriate files?
   - Answer: y
9. Is the PR name meaningful and descriptive?
   - Answer: y
10. Is PR description provided, or unnecessary?
    - Answer:  y
  
Other questions:

11. Have you not made changes to the requirements or anything related to the Singularity image? If you have, did you generate and upload a new image to the clusters?
    - Answer: n
12. Is the change significant enough to notify all individuals working with this repository?
    - Answer: y
13. Is there a need for a second reviewer?
    - Answer: n

### CODE REVIEW

**Comment by ludziej** [comment]
Test aktualnie przechodzi w kilkanaście sekund i uruchamia się za każdym commitem, jeśli to za długo to dam mu supress dopóki testy nie będą na hooks. Jest narazie jeden prosty baseline test [research/conditional/train/configs/test/test_baseline.yaml](https://github.com/llm-random/llm-random/pull/192/files#diff-bf719f54f0de393da5db3b222a5b880665c5cb0559c2f69580c2c7073edd313d), dowolny yaml w tym samym katalogu będzie uznany z testowy.

**Comment by ludziej** [comment]
[Neptune project](https://app.neptune.ai/o/pmtest/org/llm-random-tests/runs/details?viewId=standard-view&detailsTab=metadata&shortId=LLMTST-2&path=&attribute=loss) z testowymi runami.

**Review by sebastianjaszczur — APPROVED** [review]
Nice, thanks! LGTM, only minor changes suggested. In general it is so much better to have any integration test, that we should merge even this relatively big model with unoptimized dataset (altho the former is easy to fix)

File: research/conditional/train/configs/test/test_baseline.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  batch_size: 16
+  cutoff: 128
+  project_name: "pmtest/llm-random-tests"
+  name: "baseline_test"
+  mixed_precision: true
+  tags:
+    - "test"
+  use_neptune: true
+  n_steps: 10
+  dmodel: 256
```
[the remark this answers is not in the document]
Comment (ludziej) [reply research/conditional/train/configs/test/test_baseline.yaml:19]: I've reduced it, but still loading the dataset is the bottleneck.

**Comment by sebastianjaszczur** [comment]
Wgl dobry design kodu, podoba mi się jak to się odpala. Dzięki! Wgl dodałem task na Trello odnośnie zmniejszenia datasetu w integration test.

**Pull request merged**